### PR TITLE
3DS2 Initiate payment request

### PR DESCRIPTION
**Pull request by maassenbas**

### CODE REVIEW

File: cartridges/int_adyen_overlay/cartridge/scripts/util/AdyenHelper.ds [hunk 1]
```diff
@@ -138,7 +146,7 @@ var __AdyenHelper: Object = {
     },
 
     getCheckoutUrl: function (): String {
-        var checkoutUrl = this.getLoadingContext() + "sdk/2.1.0/adyen.js";
+        var checkoutUrl = this.getLoadingContext() + "sdk/2.2.1/adyen.js";
```
Comment (rikterbeek) [inline cartridges/int_adyen_overlay/cartridge/scripts/util/AdyenHelper.ds]: Do not use this version please upgrade to v2.5.0

File: cartridges/int_adyen_overlay/cartridge/scripts/util/AdyenHelper.ds [hunk 2]
```diff
@@ -446,6 +454,24 @@ var __AdyenHelper: Object = {
...
+        var card;
+        if (session.forms.billing.paymentMethods) {
+            card = session.forms.billing.paymentMethods.creditCard;
+        }
+        else {
+            card = session.forms.billing.creditCardFields;
+        }
+
+        jsonObject["browserInfo"] = card.browserInfo.value;
+        jsonObject.threeDS2RequestData = {"notificationURL" : "https://test.com"}
```
Comment (rikterbeek) [inline cartridges/int_adyen_overlay/cartridge/scripts/util/AdyenHelper.ds]: do not specify a notificationURL this is being handled by the component.

File: cartridges/int_adyen_overlay/cartridge/scripts/util/AdyenHelper.ds [hunk 3]
```diff
@@ -446,6 +454,24 @@ var __AdyenHelper: Object = {
...
+
+        var card;
+        if (session.forms.billing.paymentMethods) {
+            card = session.forms.billing.paymentMethods.creditCard;
+        }
+        else {
+            card = session.forms.billing.creditCardFields;
+        }
+
+        jsonObject["browserInfo"] = card.browserInfo.value;
```
Comment (rikterbeek) [inline cartridges/int_adyen_overlay/cartridge/scripts/util/AdyenHelper.ds]: I believe some of the browserInfo data needs to be retrieved from the backend. @cyattilakiss do we have docs on this part ?

File: cartridges/int_adyen_SFRA/cartridge/templates/default/account/payment/paymentForm.isml [hunk 4]
```diff
@@ -1,7 +1,6 @@
 <isscript>
     var assets = require('*/cartridge/scripts/assets.js');
     assets.addCss('/css/adyenCss.css');
-    assets.addJs('/js/adyenSecureFields.js');
     assets.addCss('/css/adyen-checkout-2.1.0.css');
```
Comment (Aleffio) [inline cartridges/int_adyen_SFRA/cartridge/templates/default/account/payment/paymentForm.isml]: please update the css as well to 2.5.0

File: cartridges/int_adyen_overlay/cartridge/scripts/util/AdyenHelper.ds [hunk 5]
```diff
@@ -446,6 +454,24 @@ var __AdyenHelper: Object = {
         return {error: true};
     },
 
+    add3DS2Data: function(jsonObject){
+        jsonObject["additionalData"].allow3DS2 = true;
+        jsonObject["channel"] = "web";
+
```
Comment (Aleffio) [inline cartridges/int_adyen_overlay/cartridge/scripts/util/AdyenHelper.ds]: please add 'origin' to the request fields: https://github.com/Adyen/adyen-magento/blob/feature/PW-1146-3DS2/app/code/community/Adyen/Payment/Model/Api.php#L147

File: cartridges/int_adyen_overlay/cartridge/scripts/util/AdyenHelper.ds (hunk 3 above)
Comment (cyattilakiss) [reply cartridges/int_adyen_overlay/cartridge/scripts/util/AdyenHelper.ds]: https://github.com/Adyen/adyen-magento2/pull/447/files#diff-aba9447e7c2f9df9662124e47fd30e2aR275
These 2 only, but it should be already in the codebase.

File: metadata/site_import/meta/system-objecttype-extensions.xml [hunk 6]
```diff
@@ -264,6 +264,13 @@
 
     <type-extension type-id="SitePreferences">
         <custom-attribute-definitions>
+            <attribute-definition attribute-id="Adyen3DS2Enabled">
```
Comment (cyattilakiss) [inline metadata/site_import/meta/system-objecttype-extensions.xml]: If you set this, is it going to be shown on the admin of the platform? Can we hide it?

Comment (rikterbeek) [reply metadata/site_import/meta/system-objecttype-extensions.xml]: showing it is fine as they should be able to turn it on.

Comment (maassenbas) [reply metadata/site_import/meta/system-objecttype-extensions.xml]: It won't be shown in the business manager. This just creates the variable, but not displayed in custom preferences.

**Pull request merged**